common/djangoapps/track/backends/tests/test_oli.py

```diff
@@ -0,0 +1,247 @@
+"""Tests OLIAnalyticsBackend: Transmit events to the OLI analytics service"""
+from __future__ import absolute_import
+
+from mock import patch, Mock
+
+from django.test import TestCase
+from south.utils import datetime_utils as datetime
+
+from track.backends.oli import OLIAnalyticsBackend
+from student.tests.factories import UserFactory
+
+
+# pylint: disable=line-too-long
+class TestOLIBackend(TestCase):
+    """
+    Tests OLIAnalyticsBackend.  OLI should only process 'problem_check' event type.
+    Course IDs must be part of the whitelisted set
+    """
+    def setUp(self):
+        self.patcher = patch('track.backends.oli.OAuth1Session.put', mock_put)
+        self.addCleanup(self.patcher.stop)
+        self.patcher.start()
+        self.backend = OLIAnalyticsBackend()
+
+        # create a test user
+        self.user = UserFactory.create()
+        self.user.save()
+
+        # initialize attributes for the test cases.
+        self.backend.url = 'https://example.com'
+        self.backend.path = 'path'
+        self.backend.key = 'key'
+        self.backend.secret = 'secret'
+        self.backend.course_id_set = ['edX/DemoX/Demo_Course', 'edX/DemoX/Demo_Course1']
```

Comment:
The object initializes this as a `set` but here in the tests we're
assigning it to a `list`. This should look like:
```python
self.backend.course_ids = {
    'edX/DemoX/Demo_Course',
    'edX/DemoX/Demo_Course1',
}
```